File: samplejava/src/main/java/com/kazakago/factolize/samplejava/SubFragment.java
package com.kazakago.factolize.samplejava;

import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.kazakago.factolize.Factory;
import com.kazakago.factolize.FactoryParam;

@Factory
public class SubFragment extends Fragment {

    @FactoryParam
    int intValue;
    @FactoryParam
    String stringValue;

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        SubFragmentFactory.injectArgument(this, savedInstanceState);
    }

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        return inflater.inflate(R.layout.fragment_sub, container, false);
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        TextView intValueTextView = view.findViewById(R.id.intValueTextView);
        intValueTextView.setText("intValue : " + intValue);
        TextView stringValueTextView = view.findViewById(R.id.stringValueTextView);
        stringValueTextView.setText("stringValue" + stringValue);
    }
}

File: samplekotlin/src/main/java/com/kazakago/factolize/samplekotlin/SubFragment.kt
package com.kazakago.factolize.samplekotlin

import android.arch.lifecycle.Observer
import android.arch.lifecycle.ViewModelProviders
import android.os.Bundle
import android.support.v4.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.kazakago.factolize.Factory
import com.kazakago.factolize.FactoryParam
import kotlinx.android.synthetic.main.fragment_sub.*

@Factory
class SubFragment : Fragment() {

    @FactoryParam
    var intValue: Int = 0
    @FactoryParam
    var stringValue: String = ""

    private lateinit var viewModel: SubViewModel

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        SubFragmentFactory.injectArgument(this)

        val viewModelFactory = SubViewModelFactory(activity!!.application, intValue, stringValue)
        viewModel = ViewModelProviders.of(this, viewModelFactory).get(SubViewModel::class.java)
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        return inflater.inflate(R.layout.fragment_sub, container, false)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        viewModel.intValueLiveData.observe(this, Observer {
            intValueTextView.text = String.format("intValue : %d", it)
        })
        viewModel.stringValueLiveData.observe(this, Observer {
            stringValueTextView.text = String.format("stringValue : %s", stringValue)
        })
    }

}
